src/PowerShellEditorServices/Language/FindDeclarationVisitor.cs

```diff
@@ -76,19 +78,77 @@ public override AstVisitAction VisitFunctionDefinition(FunctionDefinitionAst fun
         /// or a decision to continue if it wasn't found</returns>
         public override AstVisitAction VisitAssignmentStatement(AssignmentStatementAst assignmentStatementAst)
         {
-            var variableExprAst = assignmentStatementAst.Left as VariableExpressionAst;
-            if (variableExprAst == null ||
-                variableName == null ||
-                !variableExprAst.VariablePath.UserPath.Equals(
-                    variableName,
-                    StringComparison.OrdinalIgnoreCase))
+            if (variableName == null)
             {
                 return AstVisitAction.Continue;
             }
 
-            // TODO also find instances of set-variable
-            FoundDeclaration = new SymbolReference(SymbolType.Variable, variableExprAst.Extent);
-            return AstVisitAction.StopVisit;
+            // We want to check VariableExpressionAsts from within this AssignmentStatementAst so we visit it.
+            FindDeclarationVariableExpressionVisitor visitor = new FindDeclarationVariableExpressionVisitor(symbolRef);
+            assignmentStatementAst.Visit(visitor);
```

Comment:
Is it possible to make this `assignmentStatementAst.Left.Visit(visitor)` and drop the other methods in `FindDeclarationVariableExpressionVisitor`?

Reply:
Also if you go this route, add an override for `VisitMemberExpression` and have it return skip children.

Reply:
Actually also `VisitIndexExpression`